### PR TITLE
20 Done the reducer thing

### LINKED ISSUE #20: Update the info reducer and category reducer with getCategory action types
The info reducer will be in `reducers/info.js` and will need to handle the `GET_CATEGORY_PENDING` action type

The category reducer will be in `reducers/foods.js` and will need to handle the `GET_CATEGORY_SUCCESS` action type

Use the existing switch statement in `reducers/info.js` as a guide - the functionality should be very similar.

### PR DESCRIPTION
**Pull request by Rahul-Lal**
Can you check?

Closes #20

### CODE REVIEW

**Review by emilyparkes — CHANGES_REQUESTED** [review]
Please mention `closes #[ticketnumberhere]` in PR description

File: client/reducers/foods.js [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
       return action.foods
+    default:
+      return state
+  }
+}
 
+export default function categoryReducer (state = {}, action) {
+  switch (action.type) {
+    case GET_CATEGORIES_SUCCESS:
+      return action.categories
```
Comment (emilyparkes) [inline client/reducers/foods.js]: This should be returning the one ✅ `category` not the list of 🚫 `categories`.

File: client/reducers/foods.js [hunk 2]
```diff
@@ -1,10 +1,19 @@
...
     case GET_FOODS_SUCCESS:
       return action.foods
+    default:
+      return state
+  }
+}
 
+export default function categoryReducer (state = {}, action) {
+  switch (action.type) {
+    case GET_CATEGORIES_SUCCESS:
```
Comment (emilyparkes) [inline client/reducers/foods.js]: This should be ✅ `GET_CATEGORY_SUCCESS` not  🚫 `GET_CATEGORIES_SUCCESS`.

**Review by emilyparkes — CHANGES_REQUESTED** [review]

File: client/reducers/foods.js [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
 export default function foodReducer (state = [], action) {
   switch (action.type) {
     case GET_FOODS_SUCCESS:
       return action.foods
+    default:
+      return state
+  }
+}
 
+export default function categoryReducer (state = {}, action) {
```
Comment (emilyparkes) [inline client/reducers/foods.js]: Please write a test for this reducer too in `tests/client/reducer/foods.test.js`

File: client/reducers/foods.js [hunk 4]
```diff
@@ -1,10 +1,19 @@
 import { GET_FOODS_SUCCESS } from './../actions/foods'
+import { GET_CATEGORIES_SUCCESS } from './../actions/categories'
```
Comment (bryceknz) [inline client/reducers/foods.js]: You should be using the action type from `/actions/category`, not `/actions/categories`.

File: client/reducers/foods.js (hunk 3 above)
Comment (bryceknz) [inline client/reducers/foods.js]: Is it not possible to have two `default` exports in one file. I suspect your intent was actually just to add to the `switch` statement the following:
```js
case GET_CATEGORY_SUCCESS:
  return action.category
```

File: tests/client/reducers/foods.tests.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { GET_CATEGORY_SUCCESS } from '..//../../client/actions/category'
+import category from '../../../client/reducers/foods'
+
+describe('jest', () => {
+  it('jest is working', () => {
```
Comment (bryceknz) [inline tests/client/reducers/foods.tests.js]: @Rahul-Lal, you and I both know that this isn't what @emilyparkes meant when she said to include a test 😉

**Review by bryceknz — DISMISSED** [review]
There are no changes to the `info` reducer to handle the `getCategory` action types. Confusion around `getCategory` and `getCategories` is likely due to unclear guidance given, I apologise for that.
You will need to:
- [ ] add handling for `GET_CATEGORY_PENDING` etc. into the `info` reducer
- [ ] include tests for both of these reducers (look at `/tests/client/reducers/info.js` for guidance).

File: client/reducers/foods.js (hunk 4 above)
Comment (emilyparkes) [inline client/reducers/foods.js]: You should be using the action type from `/actions/category`, not `/actions/categories`.

File: client/reducers/foods.js (hunk 3 above)
Comment (emilyparkes) [inline client/reducers/foods.js]: Is it not possible to have two `default` exports in one file. All that is required for this ticket is to add the following to the `switch` statement:
```js
case GET_CATEGORY_SUCCESS:
  return action.category
```

File: tests/client/reducers/foods.tests.js (hunk 5 above)
Comment (emilyparkes) [inline tests/client/reducers/foods.tests.js]: @Rahul-Lal, this needs more in depth testing please!

**Review by emilyparkes — CHANGES_REQUESTED** [review]

**Review by emilyparkes — APPROVED** [review]

**Pull request merged**